src/energies.c

```diff
@@ -177,16 +177,17 @@ double internal_loop_energy(int i, int k, int l, int j, plain_sequence *rna){
 double hairpin_energy(int i, int j, plain_sequence * rna){
   double hairpin_energy;
   int hsize, ptype;
-  char *hseq; /* sequence of unpaired bases in hairpin */
+  char hseq[10]; /* sequence of unpaired bases in hairpin */
   hsize = j - i - 1;
   ptype = get_type(rna->label[i], rna->label[j]);
-  hseq = (char*) malloc ((hsize+1)*sizeof(char));
-  strncpy(hseq, rna->label+i+1, hsize);
-  hseq[hsize] = '\0';
+  hseq[0] = '\n';
```

Comment:
This should actually be '\0'...

Reply:
Hello, RaumZeit,
thanks for your explanation and fix!